test/browser.test.js

```diff
@@ -119,18 +116,19 @@ function browserifyTestApp(options, strategy, next) {
   var appDir = typeof(options) === 'object' ? options.appRootDir : options;
   var b = compileStrategies[strategy](appDir);
 
-  boot.compileToBrowserify(options, b);
-
-  exportBrowserifyToFile(b, 'browser-app-bundle.js', next);
+  boot.compileToBrowserify(options, b, function(err) {
+    exportBrowserifyToFile(b, 'browser-app-bundle.js', next);
+  });
 }
 
-function executeBundledApp(bundlePath) {
+function executeBundledApp(bundlePath, done) {
   var code = fs.readFileSync(bundlePath);
   var context = createBrowserLikeContext();
   vm.runInContext(code, context, bundlePath);
   var app = vm.runInContext('require("browser-app")', context);
-
-  printContextLogs(context);
-
+  app.start(function(err) {
+    printContextLogs(context);
+    done(err);
+  });
   return app;
```

Comment:
An async function that also returns a value in sync way is very weird. Please pass `app` to the callback.

```js
app.start(function(err) {
  printContextLogs(context);
  done(err, app);
});
```